alws/crud/errata.py

```diff
@@ -156,6 +161,148 @@ async def get_oval_xml(
+                ),
+                "advisory": {
+                    "from": record.contact_mail,
+                    "severity": record.severity.capitalize(),
+                    "rights": record.rights,
+                    "issued_date": record.issued_date,
+                    "updated_date": record.updated_date,
+                    "affected_cpe_list": record.affected_cpe,
+                    "bugzilla": [
+                        {
+                            "id": ref.ref_id,
+                            "href": ref.href,
+                            "title": ref.title,
+                        }
+                        for ref in record.references
+                        if ref.ref_type == ErrataReferenceType.bugzilla
+                    ],
+                    "cves": [
+                        {
+                            "name": ref.ref_id,
+                            "public": datetime.datetime.strptime(
+                                # year-month-day
+                                ref.cve.public[:10],
+                                "%Y-%m-%d",
+                            ).date(),
+                            "href": ref.href,
+                            "impact": ref.cve.impact.capitalize(),
+                            "cwe": ref.cve.cwe,
+                            "cvss3": ref.cve.cvss3,
+                        }
+                        for ref in record.references
+                        if ref.ref_type == ErrataReferenceType.cve and ref.cve
+                    ],
+                },
+                "references": [
+                    {
+                        "id": ref.ref_id,
+                        "url": ref.href,
+                        "source": (
+                            "RHSA"
+                            if ref.ref_type == ErrataReferenceType.rhsa
+                            else "ALSA"
+                        )
+                    }
+                    for ref in record.references
+                    if ref.ref_type in [
+                        ErrataReferenceType.self_ref, ErrataReferenceType.rhsa
+                    ]
+                ],
+            },
+            "criteria": record.criteria,
+        })
+        oval.append_object(definition)
+        for test in record.tests:
+            if test["id"] not in objects:
+                objects.add(test["id"])
+                oval.append_object(
+                    get_test_cls_by_tag(test["type"]).from_dict(test)
+                )
+        for obj in record.objects:
+            if obj["id"] not in objects:
+                objects.add(obj["id"])
+                oval.append_object(
+                    get_object_cls_by_tag(obj["type"]).from_dict(obj)
+                )
+        for state in record.states:
+            if state["id"] not in objects:
+                objects.add(state["id"])
+                oval.append_object(
+                    get_state_cls_by_tag(state["type"]).from_dict(state)
+                )
```

Comment:
I suppose that we can unite these loops into one. 
Something like that:
```python
for coll in (record.tests, record.objects, record.states, record.variables):
    if coll is None:
        continue
    for el in coll:
        if el['id'] in objects:
            continue
        objects.add(el["id"])
        oval.append_object(
            get_state_cls_by_tag(el["type"]).from_dict(el)
        )
```
but, it's up to you

Reply:
Yes, I see your point and I actually considered it when implementing it, but every oval ref has its own `get_*_cls_by_tag`, hence I decided to keep it as different loops

Reply:
oh, I see that now, but still, we can add these functions to our tuple like that:
```python
for coll, func in (
    (record.tests, get_test_cls_by_tag),
):
    ...
    for el in coll:
        oval.append_object(func(...).from_dict(el))
```

Reply:
Ok, turned oval objects addition into a function and followed your loop nesting approach in https://github.com/AlmaLinux/albs-web-server/pull/998/commits/d36ac66ce566bd1c25486a48289f9b895b080dd2, let me know what do you think